src/main/resources/openapi/NotificationsAPIs.yaml

```diff
@@ -1075,6 +1075,75 @@ paths:
         connectionId: '${vpc_connection_id}'
         httpMethod: POST
 
+  #
+  # Generate bespoke comm Template Preview
+  # --------------------------------------------------------------------------------
+
+  '/templates/bespoke-comm/preview':
+    options:
+      summary: CORS support
+      description: |
+        Enable CORS by returning correct headers
+      tags:
+        - Generate bespoke comm matched Template Preview
+      responses:
+        200:
+          description: Default response for CORS method
+          headers:
+            Access-Control-Allow-Origin:
+              schema:
+                type: string
+            Access-Control-Allow-Methods:
+              schema:
+                type: string
+            Access-Control-Allow-Headers:
+              schema:
+                type: string
+          content: { }
+      x-amazon-apigateway-integration:
+        type: mock
+        requestTemplates:
+          application/json: |
+            {
+              "statusCode" : 200
+            }
+        responses:
+          default:
+            statusCode: "200"
+            responseParameters:
+              method.response.header.Access-Control-Allow-Headers: '''Content-Type,X-Amz-Date,Authorization,X-Api-Key'''
+              method.response.header.Access-Control-Allow-Methods: '''*'''
+              method.response.header.Access-Control-Allow-Origin: '''*'''
+            responseTemplates:
+              application/json: |
+                {}
+    post:
+      summary: Generates a text preview for bespoke comm template type and personalisation data
+      description: Generates a text preview for bespoke comm template type and personalisation data
+      operationId: generate-bespoke-comm-template-preview
+      tags:
+        - Generate Bespoke Comm Template Preview
+      requestBody:
+        $ref: '#/components/requestBodies/GenerateBespokeCommTemplatePreview'
+      responses:
+        '201':
+          $ref: '#/components/responses/GenerateTemplatePreview'
+        '400':
```

Comment:
It looks like there's also a 404 response from looking at the tests and the GovNotifyApiClient